src/mrpro/operators/_SliceProjectionOp.py

```diff
@@ -0,0 +1,449 @@
+        n_batchdim = len(self._range_shape) - 3
+        # x_domainbatch_range has all volume batch dimensions moved to the front
+        x_domainbatch_range = x.moveaxis(tuple(range(n_batchdim, x.ndim - 3)), tuple(range(x.ndim - 3 - n_batchdim)))
+        # x_flatdomainbatch_flatrange is 2D with shape
+        # (all batch dimensions of the volume flattened, all range dimensions flattened)
+        x_domainbatch_flatrange = torch.atleast_2d(x_domainbatch_range.flatten(start_dim=-len(self._range_shape)))
+        x_flatdomainbatch_flatrange = x_domainbatch_flatrange.flatten(end_dim=-2)
+        # y_flatdomainbatch has shape (all batch dimensions of the volume flattened, *range dimensions)
+        y_flatdomainbatch = torch.stack(
+            [
+                _MatrixMultiplication.apply(x, matrix_adjoint, matrix).reshape(self._domain_shape)
+                for x in x_flatdomainbatch_flatrange
+            ]
+        )
+        y = y_flatdomainbatch.reshape(*x.shape[n_batchdim:-3], *y_flatdomainbatch.shape[1:])
+        return (y,)
+
+    @staticmethod
+    def join_matrices(matrices: Sequence[torch.Tensor]) -> torch.Tensor:
+        values = []
+        target = []
+        source = []
+        for i, m in enumerate(matrices):
+            if not m.shape == matrices[0].shape:
+                raise ValueError('all matrices should have the same shape')
+            c = m.coalesce()
+            (ctarget, csource) = c.indices()
+            values.append(c.values())
+            source.append(csource)
+            ctarget = ctarget + i * m.shape[0]
+            target.append(ctarget)
+
+        with warnings.catch_warnings():
+            # beta status in pytorch causes a warning to be printed
+            warnings.filterwarnings('ignore', category=UserWarning, message='Sparse')
+            matrix = torch.sparse_coo_tensor(
+                indices=torch.stack([torch.cat(target), torch.cat(source)]),
+                values=torch.cat(values),
+                dtype=torch.float32,
+                size=(len(matrices) * m.shape[0], m.shape[1]),
+            )
+        return matrix
+
+    @staticmethod
+    def projection_matrix(
+        input_shape: SpatialDimension[int],
+        output_shape: SpatialDimension[int],
+        rotation: Rotation,
+        offset: Tensor,
+        w: int,
+        slice_function: TensorFunction,
+        rotation_center: torch.Tensor | None = None,
+    ) -> torch.Tensor:
+        """Create a sparse matrix that represents the projection of a volume onto a plane.
+
+        Outside the volume values are approximately zero padded
+
+        Parameters
+        ----------
+        input_shape:
+            Shape of the volume to sample from
+        output_shape:
+            Shape of the resulting plane, 2D.
+        rotation
+            Rotation that describes the orientation of the plane
+        offset: Tensor
+            Offset of the plane in the volume in plane coordinates from the center of the volume
```

Comment:
what do you mean here with "plane coordinates"? I would expect that you first rotate and then shift. Is the shift in voxel coordinates? If so, is a offset =  1 shifted by one voxel of the 3D volume or by the slice thickness, i.e. fwhm?

Reply:
voxel coordinates.
does this make sense (and I only add it to the docstring) or should it be different?

What is meant here is:
"Shift of the plane from the center of the volume, perpendicular the the plane, in units of the 3D volume."

Changing the pixel/voxelsize or shifting the plane in other direction is currently not supported in this version.

Reply:
> "Shift of the plane from the center of the volume, perpendicular the the plane, in units of the 3D volume."

sounds like a good explaination to me